crates/sui-core/src/unit_tests/move_integration_tests.rs

```diff
@@ -1918,6 +1921,60 @@ async fn test_object_no_id_error() {
                  && err_str.contains("First field of struct NotObject must be 'id'"));
 }
 
+#[tokio::test]
+#[cfg_attr(msim, ignore)]
+async fn test_generate_lock_file() {
+    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    path.extend(["src", "unit_tests", "data", "generate_move_lock_file"]);
+
+    let lock_file =
+        tempfile::NamedTempFile::new().expect("Could not create temp file for Move.lock");
+    let lock_file_path = lock_file.path().to_path_buf();
```

Comment:
One way you could avoid the confusing behaviour of having to re-open the file to read it is to use a `tempdir` instead of a tempfile, something like this:

```suggestion
    let tmp = tempfile::tempdir().expect("Could not create temp dir for Move.lock");
    let lock_file_path = tmp.path().join("Move.lock");
```

Then it's impossible to make the mistake of reading the stale file.

Reply:
good call 💃